### ISSUE TITLE
Warning about currenttemp rate when updating rig

### ISSUE THREAD

**Issue by bdr1177** [openaps/oref0#720]
Reinstalled 0.6.0 dev and got the following error: `Warning: currenttemp rate 0.4 != lastTemp rate 0.3 from pumphistory; setting neutral temp of 0.3.. Setting neutral temp basal of 0.3U/hr`  An openaps-set temp was running at the time that I reinstalled and that seemed to start off the errors.  Despite the errors, the rig seemed to be successful at setting new temps.

**Comment by scottleibrand** [comment]
Did you keep getting the same warning repeatedly? Or just once and then it went back to looping normally?

**Comment by bdr1177** [comment]
The warning has been repeating for the past two hours ever since I did a clean install of 0.6.0 dev.

**Comment by scottleibrand** [comment]
I'm confused. If it's giving you that error it can't set temps normally. Has it only been setting neutral temps this whole time? You said it also set a zero temp...

**Comment by scottleibrand** [comment]
Can you post the contents of settings/temp_basal.json and the most recent snippet of pumphistory.json?

**Comment by bdr1177** [comment]
```
Oct 15 19:35:52 natopenaps pump-loop.log: Autosens ratio: 1; Basal unchanged: 0.3; target_bg unchanged: 130; ISF unchanged: 190
Oct 15 19:35:52 natopenaps pump-loop.log: currenttemp: { duration: 25, rate: 0.3, temp: 'absolute' } lastTempAge: 7 m tempModulus: 2 m
Oct 15 19:35:52 natopenaps pump-loop.log: determine-basal://text/shell/enact/suggested.json
Oct 15 19:35:52 natopenaps pump-loop.log: reporting enact/suggested.json
Oct 15 19:35:52 natopenaps pump-loop.log: {"temp":"absolute","deliverAt":"2017-10-15T23:35:52.250Z","reason":"Warning: currenttemp rate 0.3 != lastTemp rate 0 from pumphistory; setting neutral temp of 0.3. 25m left and 0.3 ~ req 0.3U/hr: no temp required"}
Oct 15 19:35:52 natopenaps pump-loop.log: enact/enacted.json: {"received":true,"temp":"absolute","timestamp":"2017-10-15T19:29:50.327359","rate":0.3,"reason":"Warning: currenttemp rate 0 != lastTemp rate 0.3 from pumphistory; setting neutral temp of 0.3.. Setting neutral temp basal of 0.3U/hr","duration":30,"deliverAt":"2017-10-15T23:29:47.211Z"}
Oct 15 19:35:52 natopenaps pump-loop.log: Settings less than 10 minutes old. Edison on battery: 50%. Pumphistory-24 < 30m old
Oct 15 19:35:52 natopenaps pump-loop.log: Completed pump-loop at Sun Oct 15 19:35:52 EDT 2017
Oct 15 19:36:02 natopenaps pump-loop.log: Starting pump-loop at Sun Oct 15 19:36:02 EDT 2017:
Oct 15 19:36:02 natopenaps pump-loop.log: grep: enact/smb-suggested.json: No such file or directory
Oct 15 19:37:13 natopenaps pump-loop.log: Waiting up to 4 minutes for new BG: .......glucose.json newer than pump_loop_completed
Oct 15 19:37:56 natopenaps pump-loop.log: Radio ok. Listening: .No pump comms detected from other rigs
Oct 15 19:38:19 natopenaps pump-loop.log: Profile less than 60m old. glucose.json newer than temp_basal.json. Temp refreshed
Oct 15 19:38:24 natopenaps pump-loop.log: {"carbs":55,"mealCOB":23,"currentDeviation":5.28,"maxDeviation":13.19,"minDeviation":7.46,"slopeFromMaxDeviation":-0.903,"slopeFromMinDeviation":0,"allDeviations":[7.46,10.08,12.95,13.18,11.09,10.66,11.9,13.04,13.19],"lastCarbTime":1508109734000}
Oct 15 19:38:24 natopenaps pump-loop.log: {"iob":0.755,"activity":0.0039,"bolussnooze":0,"basaliob":0.755,"netbasalinsulin":0.8,"hightempinsulin":1.2,"microBolusInsulin":0.8,"microBolusIOB":0.726,"time":"2017-10-15T23:38:07.000Z","lastBolusTime":1508109734000,"lastTemp":{"rate":0.3,"timestamp":"2017-10-15T19:29:40-04:00","started_at":"2017-10-15T23:29:40.000Z","date":1508110180000,"duration":8.64}}
Oct 15 19:38:24 natopenaps pump-loop.log: {"delta":-1,"glucose":179,"short_avgdelta":0.56,"long_avgdelta":5.35}
Oct 15 19:38:24 natopenaps pump-loop.log: Autosens ratio: 1; Basal unchanged: 0.3; target_bg unchanged: 130; ISF unchanged: 190
Oct 15 19:38:24 natopenaps pump-loop.log: currenttemp: { duration: 22, rate: 0.3, temp: 'absolute' } lastTempAge: 9 m tempModulus: 1 m
Oct 15 19:38:24 natopenaps pump-loop.log: min_bg unchanged: 125; target_bg unchanged: 130; max_bg unchanged: 135
Oct 15 19:38:24 natopenaps pump-loop.log: Last carbs 16 minutes ago; remainingCATime: 3.4 hours; 58% carbs absorbed
Oct 15 19:38:24 natopenaps pump-loop.log: Carb Impact: 2.7 mg/dL per 5m; CI Duration: 3.4 hours; remaining CI (~2h peak): 2.7 mg/dL per 5m
Oct 15 19:38:24 natopenaps pump-loop.log: predCIs (mg/dL/5m): 2.6 2.6 2.5 2.4 2.4 2.3 2.2 2.2 2.1 2 2 1.9 1.8 1.8 1.7 1.6 1.6 1.5 1.4 1.4 1.3 1.2 1.2 1.1 1 1 0.9 0.8 0.8 0.7 0.6 0.6 0.5 0.5 0.4 0.3 0.3 0.2 0.1 0.1 0 0 0 0 0 0 0 0
Oct 15 19:38:24 natopenaps pump-loop.log: remainingCIs: 0.1 0.3 0.4 0.5 0.7 0.8 0.9 1 1.2 1.3 1.4 1.6 1.7 1.8 2 2.1 2.2 2.3 2.5 2.6 2.6 2.4 2.3 2.2 2.1 1.9 1.8 1.7 1.5 1.4 1.3 1.1 1 0.9 0.8 0.6 0.5 0.4 0.2 0.1 0 0 0 0 0 0 0 0
Oct 15 19:38:24 natopenaps pump-loop.log: UAM Impact: 2.7 mg/dL per 5m; UAM Duration: 0.3 hours
Oct 15 19:38:24 natopenaps pump-loop.log: minPredBG: 97 minIOBPredBG: 51 minCOBPredBG: 144 avgPredBG: 97 COB: 23 / 55
Oct 15 19:38:24 natopenaps pump-loop.log: BG projected to remain above 125 for 240 minutes
Oct 15 19:38:24 natopenaps pump-loop.log: naive_eventualBG: 36 bgUndershoot: 46.5 zeroTempDuration: 240 zeroTempEffect: 228 carbsReq: -47
Oct 15 19:38:24 natopenaps pump-loop.log: determine-basal://text/shell/enact/suggested.json
Oct 15 19:38:24 natopenaps pump-loop.log: reporting enact/suggested.json
Oct 15 19:38:24 natopenaps pump-loop.log: {"temp":"absolute","bg":179,"tick":-1,"eventualBG":144,"snoozeBG":97,"insulinReq":0,"reservoir":null,"deliverAt":"2017-10-15T23:38:24.319Z","predBGs":{"IOB":[179,178,176,174,171,168,165,161,157,153,148,144,139,134,130,125,121,117,113,109,105,101,98,94,91,88,85,82,79,77,74,72,70,68,66,64,63,61,60,58,57,56,55,54,53,52,52,51],"aCOB":[179,185,190,194,198,201,204,206,207,208,209,209,208,208,207,206,204,202,200,197,194,191,187,184,180,177,174,171,169,166,164,162,159,157,156,154,152,150,149,148,147,145,144,143,143,142,141],"COB":[179,178,177,176,174,172,171,169,167,166,164,163,162,160,159,159,158,158,157,157,157,158,158,158,158,158,158,158,158,157,157,157,156,156,155,155,154,153,152,151,150,149,148,147,146,145,145,144]},"COB":23,"IOB":0.755,"reason":"COB: 23, Dev: 16, BGI: -3.7, ISF: 190, Target: 130, minPredBG 97, minGuardBG 51, IOBpredBG 51, COBpredBG 144; minGuardBG 51<83","duration":30,"rate":0}
Oct 15 19:38:30 natopenaps pump-loop.log: enact/enacted.json: {"insulinReq":0,"received":true,"bg":179,"reservoir":null,"temp":"absolute","snoozeBG":97,"timestamp":"2017-10-15T19:38:29.862335","predBGs":{"COB":[179,178,177,176,174,172,171,169,167,166,164,163,162,160,159,159,158,158,157,157,157,158,158,158,158,158,158,158,158,157,157,157,156,156,155,155,154,153,152,151,150,149,148,147,146,145,145,144],"aCOB":[179,185,190,194,198,201,204,206,207,208,209,209,208,208,207,206,204,202,200,197,194,191,187,184,180,177,174,171,169,166,164,162,159,157,156,154,152,150,149,148,147,145,144,143,143,142,141],"IOB":[179,178,176,174,171,168,165,161,157,153,148,144,139,134,130,125,121,117,113,109,105,101,98,94,91,88,85,82,79,77,74,72,70,68,66,64,63,61,60,58,57,56,55,54,53,52,52,51]},"rate":0,"IOB":0.755,"reason":"COB: 23, Dev: 16, BGI: -3.7, ISF: 190, Target: 130, minPredBG 97, minGuardBG 51, IOBpredBG 51, COBpredBG 144; minGuardBG 51<83","COB":23,"eventualBG":144,"duration":30,"tick":-1,"deliverAt":"2017-10-15T23:38:24.319Z"}
```

**Comment by bdr1177** [comment]
Above on line 5 you can see the error and then down near the bottom of what I pasted you can see it setting the zero temp basal.  It's been doing this since about 5:30.  Looking back at NS, it looks like it was mostly setting neutral basals during this time (5:30 pm - 7:30 pm), but every once in a while it would successfully set a zero temp.

**Comment by bdr1177** [comment]
And looking back more, it wasn't just zero temps that it would set.  It also occassionally set higher temps (e.g. 0.7), but they only lasted until the next loop, which threw the error and set it back to neutral.

**Comment by bdr1177** [comment]
There's no temp_basal.json in settings and the only pumphistory.json files are pumphistory-24h.json and pumphistory-24-zoned.json.  Here's what the end of the pumphistory-24h.json looks like
```
{
    "_type": "BolusWizard",
    "bg": 0,
    "bg_target_high": 135,
    "correction_estimate": 0.0,
    "unknown_byte[10]": 0,
    "_description": "BolusWizard 2017-10-14T16:38:37 head[2], body[13] op[0x5b]",
    "timestamp": "2017-10-14T16:38:37",
    "_body": "0f5028be7d0003000000000387",
    "carb_input": 15,
    "_head": "5b00",
    "unabsorbed_insulin_total": 0.0,
    "_byte[5]": 0,
    "unabsorbed_insulin_count": "??",
    "_byte[7]": 0,
    "_date": "a5a6100e11",
    "bolus_estimate": 0.3,
    "unknown_byte[8]": 0,
    "carb_ratio": 40,
    "food_estimate": 0.3,
    "bg_target_low": 125,
    "sensitivity": 190
  },
  {
    "_type": "TempBasalDuration",
    "duration (min)": 30,
    "_description": "TempBasalDuration 2017-10-14T16:34:10 head[2], body[0] op[0x16]",
    "timestamp": "2017-10-14T16:34:10",
    "_body": "",
    "_head": "1601",
    "_date": "8aa2104e11"
  },
  {
    "_type": "TempBasal",
    "temp": "absolute",
    "_description": "TempBasal 2017-10-14T16:34:10 head[2], body[1] op[0x33]",
    "timestamp": "2017-10-14T16:34:10",
    "_body": "00",
    "_head": "3300",
    "rate": 0.0,
    "_date": "8aa2104e11"
  }
```

**Comment by scottleibrand** [comment]
Working directly with @bdr1177 last night, I determined that this was a problem with the safety checks added in #702 being incompatible with the non-SMB pump-loop.  This is now fixed in #725.
